java-streams-2/src/main/java/com/baeldung/breakforeach/CustomSpliterator.java

```diff
@@ -0,0 +1,31 @@
+package com.baeldung.breakforeach;
+
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+public class CustomSpliterator extends Spliterators.AbstractSpliterator<String> {
+
+    Spliterator<String> splitr;
+    Predicate<String> predicate;
+    boolean isMatched = true;
+
+    CustomSpliterator(Spliterator<String> splitr, Predicate<String> predicate) {
+        super(splitr.estimateSize(), 0);
+        this.splitr = splitr;
+        this.predicate = predicate;
+    }
+
+    @Override
+    public boolean tryAdvance(Consumer<? super String> consumer) {
+        boolean hadNext = splitr.tryAdvance(elem -> {
+            if (predicate.test(elem)) {
```

Comment:
Should we check `isMatched` here - just in case there were calls to `tryAdvance` after the first time `tryAdvance` returns `false`?